sky/task.py

```diff
@@ -695,6 +754,47 @@ def get_cloud_to_remote_file_mounts(self) -> Optional[Dict[str, str]]:
                 d[k] = v
         return d
 
+    def to_yaml_config(self) -> Dict[str, Any]:
```

Comment:
Do we need to expose this to the user?

Reply:
Good call, agreed it's a good idea to minimize the advertised API surface. All hidden now.